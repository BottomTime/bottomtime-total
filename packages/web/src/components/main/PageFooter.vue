<template>
  <footer id="app-footer" class="footer">
    <div class="container has-text-centered">
      <p id="app-footer-copyright" class="content block">
        Copyright &copy; Chris Carleton, {{ year }}
      </p>
      <div id="app-footer-contact" class="columns">
        <div class="column is-one-quarter-tablet is-offset-one-quarter-tablet">
          <span class="icon-text">
            <span class="icon">
              <i class="fas fa-envelope"></i>
            </span>
            <span>
              <a :href="`mailto:${config.adminEmail}`">{{
                config.adminEmail
              }}</a>
            </span>
          </span>
        </div>
        <div class="column is-one-quarter-tablet">
          <RouterLink to="/privacy">Privacy Policy</RouterLink>
        </div>
      </div>
    </div>
  </footer>
</template>

<script lang="ts" setup>
import { computed } from 'vue';
import config from '@/config';

const year = computed(() => new Date().getFullYear());
</script>
