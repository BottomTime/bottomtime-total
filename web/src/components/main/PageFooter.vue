<template>
  <section class="section">
    <footer class="footer">
      <div class="container">
        <div class="content has-text-centered">
          <p>Copyright &copy; Chris Carleton, {{ year }}</p>
        </div>
      </div>
    </footer>
  </section>
</template>

<script lang="ts" setup>
import { computed } from 'vue';
const year = computed(() => new Date().getFullYear());
</script>
