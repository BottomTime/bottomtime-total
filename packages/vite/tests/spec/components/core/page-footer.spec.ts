import { mount } from '@vue/test-utils';
import PageFooter from '../../../../src/components/core/page-footer.vue';

describe('Page Footer component', () => {
  it('will render correctly', () => {
    const wrapper = mount(PageFooter);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
