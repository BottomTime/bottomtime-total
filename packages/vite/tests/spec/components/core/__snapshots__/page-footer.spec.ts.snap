// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Footer component will render correctly 1`] = `
"<section class="font-content text-grey-700 text-center h-36 pt-8">
  <p class="text-sm italic p-3">Copyright Â© Chris Carleton, 2024</p>
  <ul class="text-xs flex flex-wrap justify-center">
    <li><a href="#">Privacy Policy</a></li>
    <li class="p-0.5">|</li>
    <li><a href="#">Terms of Service</a></li>
    <li class="p-0.5">|</li>
    <li><a href="#">Cookie Policy</a></li>
    <li class="p-0.5">|</li>
    <li><a href="#">Contact</a></li>
  </ul>
</section>"
`;
